clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -621,6 +631,9 @@ private KafkaAdminClient(AdminClientConfig config,
             retryBackoffMaxMs,
             CommonClientConfigs.RETRY_BACKOFF_JITTER);
         this.clientTelemetryEnabled = config.getBoolean(AdminClientConfig.ENABLE_METRICS_PUSH_CONFIG);
+        List<MetricsReporter> reporters = CommonClientConfigs.metricsReporters(this.clientId, config);
+        this.clientTelemetryReporter = clientTelemetryReporter;
+        this.clientTelemetryReporter.ifPresent(reporters::add);
```

Comment:
Should we add close of the client telemetry reporter on admin client close?

Reply:
Added  `metrics.close()`to the `KafkaAdmin.close()` - `metrics.close()` does close all reporters

Reply:
Hmmm, but the close will not send the last telemetry push as ClientTelemetryReporter need `initiateClose` method call to set state as `terminating_push_needed` which sends the last telemetry metrics to broker. However if we decide not to send last telemetry push in admin client that's fine but we should write it in comments somewhere.

Reply:
@apoorvmittal10 good catch, I've updated the close method to call `initiateClose`

Reply:
👍